projects/packages/post-list/src/class-post-list.php

```diff
@@ -153,6 +153,53 @@ public function add_filters_and_actions_for_screen( $current_screen ) {
 
 		$this->maybe_customize_columns( $current_screen->post_type );
 		$this->maybe_add_share_action( $current_screen->post_type );
+		$this->maybe_add_copy_link_action( $current_screen->post_type );
+	}
+
+	/**
+	 * Checks the current post type and adds the Copy link post
+	 * action if it is appropriate to do so.
+	 *
+	 * @param string $post_type The post type associated with the current request.
+	 */
+	public function maybe_add_copy_link_action( $post_type ) {
+		if ( is_post_type_viewable( $post_type ) ) {
+			wp_add_inline_script(
+				'common',
+				'function copyLinkQuickAction( event ) {
+					event.preventDefault();
+					window.navigator.clipboard.writeText( event.target.getAttribute("href") ).then(() => {
+						event.target.textContent = "' . esc_js( __( 'Copied!', 'jetpack-post-list' ) ) . '";
+						setTimeout(() => {
```

Comment:
Should we clean the previous timeout just in case someone initiated this quickly before?